### PR TITLE
fvm: Engine per network

### PR DESCRIPTION
**Pull request by magik6k**
Depends on https://github.com/filecoin-project/ref-fvm/pull/494

### CODE REVIEW

**Review by Stebalien — APPROVED** [review]

File: rust/src/fvm/machine.rs [hunk 1]
```diff
@@ -31,7 +33,8 @@ pub type CgoExecutor = DefaultExecutor<
 >;
 
 lazy_static! {
-    static ref ENGINE: fvm::machine::Engine = fvm::machine::Engine::default();
+    static ref ENGINES: Mutex<BTreeMap<NetworkVersion, fvm::machine::Engine>> =
```
Comment (dignifiedquire) [inline rust/src/fvm/machine.rs:36]: why BTreeMap, is the sorting relevant?

File: rust/src/fvm/machine.rs [hunk 2]
```diff
@@ -147,8 +151,29 @@ pub unsafe extern "C" fn fil_create_fvm_machine(
         let blockstore = blockstore.finish();
 
         let externs = CgoExterns::new(externs_id);
+
+        let engine = match (|| -> anyhow::Result<Engine> {
+            let mut engines = ENGINES
+                .lock()
+                .map_err(|_| anyhow::Error::msg("engines lock is poisoned"))?;
```
Comment (dignifiedquire) [inline rust/src/fvm/machine.rs:158]: `anyhow::anyhow!` is the more common way of doing this, especially including the error would be nice if you do this

Comment (dignifiedquire) [inline rust/src/fvm/machine.rs:158]: in general there is no need to catch this, you can just unwrap it, given this indicates a previous panic, and we already catch panics

**Review by Stebalien — APPROVED** [review]
LGTM after release.

**Pull request merged**